### PR TITLE
[packages] Adding fixes to get xdg-settings working.

### PR DESCRIPTION
**Pull request by yusuf-goog**
This change sets up a .desktop file for our cipd chrome package which is required to get xdg-settings working.

Unblocks: https://github.com/flutter/packages/pull/4223

Bug:https://github.com/flutter/flutter/issues/130074

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by yusuf-goog** [comment]
@stuartmorgan if you want to merge this into yours or you can copy the code over

**Comment by yusuf-goog** [comment]
FYI: successful build with this change including @stuartmorgan code: 
https://ci.chromium.org/ui/p/flutter/builders/try/Linux_desktop%20build_all_packages%20master/853/overview

**Comment by stuartmorgan** [comment]
I'm merging it in to the main PR, since just adding a shell script that we're not using anywhere means there's no context or testing of the change.

File: .ci/scripts/set_default_linux_apps.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+set -e
+
+LOCAL_DESKTOP_FILE_DIR=~/.local/share/applications
+mkdir -p "$LOCAL_DESKTOP_FILE_DIR"
+CIPD_CHROME_DESKTOP_FILE=${LOCAL_DESKTOP_FILE_DIR}/cipd-chrome.desktop
+
+#MimeType=application/pdf;application/rdf+xml;application/rss+xml;application/xhtml+xml;application/xhtml_xml;application/xml;image/gif;image/jpeg;image/png;image/webp;text/html;text/xml;x-scheme-handler/http;x-scheme-handler/https;
+cat << EOF > "$CIPD_CHROME_DESKTOP_FILE"
```
Comment (ricardoamador) [inline .ci/scripts/set_default_linux_apps.sh:12]: nit: I am not sure what the google style doc says but in shell I prefer to use {} with variable references. So 
$CIPD_CHROME_DESKTOP_FILE should be ${CIPD_CHROME_DESKTOP_FILE}.

Comment (yusuf-goog) [reply .ci/scripts/set_default_linux_apps.sh]: Fixed

**Review by ricardoamador — APPROVED** [review]
LGTM

File: .ci/scripts/set_default_linux_apps.sh (hunk 1 above)
Comment (stuartmorgan) [reply .ci/scripts/set_default_linux_apps.sh]: > nit: I am not sure what the google style doc says but in shell I prefer to use {} with variable references.

That is the Google Bash style preference as well.

Comment (ricardoamador) [reply .ci/scripts/set_default_linux_apps.sh]: Awesome! Thanks Stuart.

**Comment by yusuf-goog** [comment]
> 

@stuartmorgan  ok, i'll close this PR then. Please pick up this latest version of the file.

**Comment by stuartmorgan** [comment]
I believe I've applied all of the missing bracing to my version. (My copy has some simplification and some added use of constants instead of repeat strings, so it's not just a case of copying it over.)

**Pull request closed without merging**